scripts/generate_node_indexes.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -eu
+
+MODULE_SRC_DIR=${1:?}
+
+find_module_files() {
+    for moduleFile in ${MODULE_FILES}; do
+        echo "${moduleFile}" | grep "^$1/[^/]*$" | sed "s,$1/,,g"
+    done
+}
+
+find_sub_dirs() {
+    for subDir in ${MODULE_DIRS}; do
+        echo "${subDir}" | grep "^$1/[^/]*$" | sed "s,$1/,,g"
+    done
+}
+
+pushd "${MODULE_SRC_DIR}" > /dev/null
+
+find . -type f \( -name '*.js' -o -name '*.d.ts' \) -exec grep -i 'GENERATED CODE -- NO SERVICES IN PROTO' {} \; -print -exec rm -f {} \;
+
+MODULE_FILES=$(find . -name '*.d.ts' -type f -print | sed 's/\.d\.ts$//g')
+MODULE_DIRS=$(find . -type d -print)
+
+for moduleDir in ${MODULE_DIRS}; do
+    pushd "${moduleDir}" > /dev/null
+
+    rm -rf index.ts
```

Comment:
I guess it doesn't hurt but this doesn't need the `-r` option as it's only supposed to be ensuring a single file doesn't exist -- my mistake